taf/utils.py

```diff
@@ -403,6 +403,19 @@ def ensure_pre_push_hook(auth_repo_path: Path) -> bool:
     return True
 
 
+def find_keystore(path: Path) -> Optional[Path]:
+    """
+    Find keystore starting from the given path and traversing parent directories if needed.
```

Comment:
I think that this function should simply call `find_taf_directory` and then check if there is a keystore directory directly inside it.